Mapping/Loader/AttributesLoader.php

```diff
@@ -54,30 +55,93 @@ public function __construct(
      * {@inheritdoc}
      */
     public function loadClassMetadata(
-        ClassMetadata $classMetadata,
+        ClassMetadataInterface $classMetadata,
         array $normalizationGroups = null,
         array $denormalizationGroups = null,
         array $validationGroups = null
     ) {
+        $this->populateFromSerializerMetadata($classMetadata, $normalizationGroups, $denormalizationGroups);
```

Comment:
The exact pattern cannot be applied here because each loader can touch to the attribute metadata. For instance the `populateSerializer()` method can add the `write` flag to an attribute but not the `read` one then the `populateIsser()` method add the `read` one.

I was think about just splitting this in 3 classes:
- Move `getOrCreateAttribute` in a dedicated factory
- Move `populateSerializer` in a `SerializerMetadataLoader` class
- Move `populateIsser` and `populateGetterAndHasser` in a `ReflectionLoader`

I will also make `ClassMetadata` and `AttributeClassMetadata` immutable (using `with*` methods).